feat(event-stream): Support block_gossip topic

samcm opened the pull request:
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**
Adds support for the new `block_gossip` event stream topic

**Which issues(s) does this PR fix?**
https://github.com/prysmaticlabs/prysm/issues/15037

**Other notes for review**
Fairly sure this is only going to fire when the block is received via gossipsub (not via api). Not sure if we want to support this or not

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have included a uniquely named [changelog fragment file](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md#maintaining-changelogmd).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

samcm committed:
feat(event-stream): Add block_gossip topic support

samcm committed:
feat(event-stream): Add block_gossip topic support

samcm committed:
feat(event-stream): Add block_gossip topic support

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/prysmaticlabs/prysm?pullRequest=15038) <br/>All committers have signed the CLA.

samcm committed:
feat: add block gossip topic support to beacon api event stream

james-prysm commented:
Thanks for this Sam, can you add some unit tests for this?

james-prysm commented:
also see a panic 

--- FAIL: FuzzValidateBeaconBlockPubSub_Phase0 (0.19s)
    --- FAIL: FuzzValidateBeaconBlockPubSub_Phase0/seed#0 (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1970508]

james-prysm commented:
looks like you also need to run the linter `bazel run //:gazelle -- fix`

samcm committed:
fix: sync_fuzz_test panic

samcm commented:
Thanks for the hints @james-prysm! I _think_ those are sorted now, but I had a little bit of trouble with the linter so let's see how the CI goes 😅

samcm committed:
fix: check for nil operationNotifier before sending block gossip

The operationNotifier was not being checked for nil before being used,
which could lead to a panic if it was not initialized. This commit adds
a nil check to prevent the panic.

james-prysm committed:
Merge branch 'develop' into develop
